ovirt/resource_ovirt_vm.go

```diff
@@ -376,6 +382,61 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	vmBuilder.Cpu(cpu)
 
+	if blockDeviceOk {
+		if storage_domain, _ := blockDevice.([]interface{})[0].(map[string]interface{})["storage_domain"]; storage_domain != "" && templateIDOK {
+
+			// Get the reference to the service that manages the storage domains
+			sdsService := conn.SystemService().StorageDomainsService()
+
+			// Find the storage domain we want to be used for virtual machine disks
+			sdsResp, err := sdsService.List().Search("name=" + storage_domain.(string)).Send()
+			if err != nil {
+				return fmt.Errorf("Failed to search storage domains, reason: %v", err)
+			}
+
+			tds, err := getTemplateDiskAttachments(templateID.(string), meta)
+			if err != nil {
+				return fmt.Errorf("Failed to get Template disks attachments: %v", err)
+			}
+
+			if storageDomains, ok := sdsResp.StorageDomains(); ok {
+				sd := storageDomains.Slice()[0]
+
+				for i, v := range tds {
+					diskIndex := i + 1
+					disk := v.MustDisk()
+					disk.SetStorageDomain(sd)
+
+					diskattachment := ovirtsdk4.NewDiskAttachmentBuilder().
+						Disk(ovirtsdk4.NewDiskBuilder().
+							Id(disk.MustId()).
+							Format(ovirtsdk4.DISKFORMAT_COW).
```

Comment:
Rather than forcing to QCOW, you can get the disk format properly.

To get all disk info, you need to get it from the disk service rather than from the VM object - like so:
```
diskService := conn.SystemService().DisksService().DiskService(disk.MustId())
fullDiskInfo := diskService.Get().MustSend().MustDisk()
log.Printf("Format: %#v", fullDiskInfo.MustFormat())
```

Reply:
Wow! This is what I needed!
Big thank you. Added auto detect of disk format.